ec2/elasticloadbalancer.py

```diff
@@ -0,0 +1,296 @@
+        raise NonRecoverableError('Health check not added to Load Balancer. '
+                                  '{0}'.format(str(e)))
+
+    ctx.logger.info(
+        'Health check added to Load Balancer {0}.'
+        .format(load_balancer.name))
+
+
+@operation
+def use_external_elb(**_):
+
+    utils.get_external_resource_id_or_raise(
+        'elb_name', ctx.target.instance)
+
+    _verify_load_balancer_exists()
+
+
+@operation
+def delete_load_balancer(**_):
+
+    if ctx.node.properties['use_external_resource']:
+        raise NonRecoverableError('Will not delete external Load Balancer. ')
+
+    ctx.logger.debug('Attempting to delete Load Balancer.')
+
+    elb_name = ctx.node.properties['elb_name']
+
+    elb_client = connection.ELBConnectionClient().client()
+
+    try:
+        lb = elb_client.get_all_load_balancers(
+            load_balancer_names=[elb_name])[0]
+        lb.delete()
+    except (boto.exception.EC2ResponseError,
+            boto.exception.BotoServerError,
+            boto.exception.BotoClientError) as e:
+        raise NonRecoverableError('Load Balancer {0} not deleted. '
+                                  .format(str(e)))
+
+    ctx.logger.info(
+        'Load Balancer {0} deleted. '
+        .format(elb_name))
+    if 'elb_name' in ctx.instance.runtime_properties:
+        ctx.instance.runtime_properties.pop('elb_name')
+
+    return True
+
+
+def _create_health_check(properties):
+
+    ctx.logger.debug('Attempting to Create Health Check.')
+
+    target = properties['target']
+
+    if 'interval' in properties:
+        interval = properties['interval']
+    else:
+        interval = constants.HEALTH_CHECK_INTERVAL
+
+    if 'healthy_threshold' in properties:
+        healthy_threshold = properties['healthy_threshold']
+    else:
+        healthy_threshold = constants.HEALTH_CHECK_HEALTHY_THRESHOLD
+
+    if 'timeout' in properties:
+        timeout = properties['timeout']
+    else:
+        timeout = constants.HEALTH_CHECK_TIMEOUT
+
+    if 'unhealthy_threshold' in properties:
+        unhealthy_threshold = properties['unhealthy_threshold']
+    else:
+        unhealthy_threshold = constants.HEALTH_CHECK_UNHEALTHY_THRESHOLD
+
+    health_check = HealthCheck(interval=interval,
+                               target=target,
+                               healthy_threshold=healthy_threshold,
+                               timeout=timeout,
+                               unhealthy_threshold=unhealthy_threshold)
+    return health_check
+
+
+def _create_load_balancer(**_):
+
+    ctx.logger.debug('Attempting to Create Load Balancer.')
+
+    listeners = ctx.node.properties['listeners']
+
+    elb_name = ctx.node.properties['elb_name']
+
+    zones = ctx.node.properties['zones']
+
+    # security_groups = ctx.node.properties['security_groups']
+
+    try:
+        elb_client = connection.ELBConnectionClient().client()
+        lb = elb_client.create_load_balancer(name=elb_name, zones=zones,
```

Comment:
have private method that creates the parameters for load balancer with dictionary which includes security groups or not depending if they exist